src/EbmlUnicodeString.cpp

```diff
@@ -169,9 +169,9 @@ filepos_t EbmlUnicodeString::ReadData(IOCallback & input, ScopeMode ReadFully)
 
   } else {
     std::string Buffer(static_cast<std::string::size_type>(GetSize()), static_cast<char>(0));
-    input.readFully(&Buffer[0], GetSize());
+    input.readFully(Buffer.data(), GetSize());
 
-    Value.SetUTF8(Buffer.c_str()); // Let conversion to std::string cut off at the first 0
+    Value.SetUTF8(Buffer); // Let conversion to std::string cut off at the first 0
```

Comment:
This change is wrong. Using `.c_str()` here is intentional as it forces the use of a function that checks for embedded `\0` characters in the buffer. It'll cut off there, which is in accordance with the Matroska specs. That's what the comment is about.

Changing it to passing the `std::string` instance instead will simply copy the whole buffer, including the `\0` character, changing the semantics. It makes a lot of the MKVToolNix test cases fail.

Please remove the corresponding commit & force-push. Thanks.

The change above is OK.

Reply:
Good thing you have tests for that.

Reply:
Yeah. One might argue that the `SetUTF8()` function should work identically in both cases, meaning it should not just copy an `std::string` but cut it to off at the first `\0`, too. In fact, that should really be the case… It just isn't due to us not implementing it after all of the changes in `UTFstring` etc.

Reply:
uhh that's quite the hack.

How about using string_view for this?

Reply:
The purpose of `SetUTF8()` is to store a copy of the string up to & excluding the first `\0` passed to it in order to make handling of [terminator characters](https://www.rfc-editor.org/rfc/rfc8794.html#terminating-elements) easier when reading files. What you pass to it doesn't really matter.

Now that we've switched to requiring C++17, we can make use of `std::string_view`, sure. That would need its own PR, though, as that does definitely not fit under "tidying up" anymore.

Reply:
Right. Different PR. I've removed this commit in the meantime.